toothpick-runtime/src/main/java/toothpick/UnScopedProviderImpl.java

```diff
@@ -44,60 +46,65 @@ public UnScopedProviderImpl(Class<?> factoryKeyClass, boolean isProviderFactoryC
 
   //we lock on the unbound provider itself to prevent concurrent usage
   //of the unbound provider (
-  public synchronized T get(Scope scope) {
-    if (instance != null) {
-      return instance;
-    }
-
-    if (providerInstance != null) {
-      if (isLazy) {
-        instance = providerInstance.get();
-        //gc
-        providerInstance = null;
+  public T get(Scope scope) {
+    lockGet.lock();
```

Comment:
Ok, but still the benchmark has shown a good improvement in perfs, probably not due to this single change.. But as stated in this article, in the end, http://stackoverflow.com/a/11821900/693752
there is also a possibility to go even further and use read and write locks in a future. And I prefer having all the code using locks all the time than a mix of locks and synchronized. 

But you are right, my first argument was wrong, it looks like it has never been truly benchmarked ???